lib/internal/repl.js

```diff
@@ -165,3 +156,60 @@ function _replHistoryMessage() {
   this._historyPrev = Interface.prototype._historyPrev;
   return this._historyPrev();
 }
+
+
+function getTMPFile(ready) {
+  var tmpPath = os.tmpdir();
+  if (!tmpPath) {
+    return ready(new Error('no tmpdir available'));
+  }
+  tmpPath = path.join(tmpPath, process.pid + '-' + 'repl.tmp');
+  fs.open(tmpPath, 'w', 0o600, function(err, fd) {
+    if (err) {
+      return ready(err);
+    }
+    return ready(null, {
+      fd: fd,
+      path: tmpPath
+    });
+  });
+}
+
+// write data, sync the fd, close it, overwrite the target
+// with a rename, open a new tmpfile
+function writeAndSwapTMP(data, tmpInfo, target, ready) {
+  debug('writing tmp file... %s', tmpInfo.path, data);
+  return fs.write(tmpInfo.fd, data, 0, 'utf8', onwritten);
```

Comment:
Added a note about this so future humans know to change this if `writeFile` gains fd support.